packages/core/test/specs/data_sources/model/conditional_variables/ComponentConditionalVariable.ts

```diff
@@ -0,0 +1,286 @@
+import { DataSourceManager, Editor } from '../../../../../src';
+import { DataVariableType } from '../../../../../src/data_sources/model/DataVariable';
+import { MissingConditionError } from '../../../../../src/data_sources/model/conditional_variables/DataCondition';
+import { ConditionalVariableType } from '../../../../../src/data_sources/model/conditional_variables/DataCondition';
+import { GenericOperation } from '../../../../../src/data_sources/model/conditional_variables/operators/GenericOperator';
+import { NumberOperation } from '../../../../../src/data_sources/model/conditional_variables/operators/NumberOperator';
+import { DataSourceProps } from '../../../../../src/data_sources/types';
+import ComponentWrapper from '../../../../../src/dom_components/model/ComponentWrapper';
+import EditorModel from '../../../../../src/editor/model/Editor';
+import { setupTestEditor } from '../../../../common';
+
+describe('ComponentConditionalVariable', () => {
+  let editor: Editor;
+  let em: EditorModel;
+  let dsm: DataSourceManager;
+  let cmpRoot: ComponentWrapper;
+
+  beforeEach(() => {
+    ({ editor, em, dsm, cmpRoot } = setupTestEditor());
+  });
+
+  afterEach(() => {
+    em.destroy();
+  });
+
+  it('should add a component with a condition that evaluates a component definition', () => {
+    const component = cmpRoot.append({
+      type: ConditionalVariableType,
+      condition: {
+        left: 0,
+        operator: NumberOperation.greaterThan,
+        right: -1,
+      },
+      ifTrue: {
+        tagName: 'h1',
+        type: 'text',
+        content: 'some text',
+      },
+    })[0];
+    expect(component).toBeDefined();
+    expect(component.get('type')).toBe(ConditionalVariableType);
+    expect(component.getInnerHTML()).toBe('<h1>some text</h1>');
+
+    const childComponent = component.components().at(0);
+    expect(childComponent).toBeDefined();
+    expect(childComponent.get('type')).toBe('text');
+    expect(childComponent.getInnerHTML()).toBe('some text');
```

Comment:
Here you're checking only the "model" of the component, let's place also a check for the "view" so we know it's rendered as expected in the canvas.

Same for the test cases below